packages/cli/src/controllers/folder.controller.ts

```diff
@@ -104,6 +106,12 @@ export class ProjectController {
 		const { projectId, folderId } = req.params;
 
 		try {
+			await this.folderService.findFolderInProjectOrFail(folderId, projectId);
+
+			if (!payload.transferToFolderId) {
```

Comment:
So with this condition here and the check for the project, I guess this will never be executed? at least in this controller?

![CleanShot 2025-05-08 at 18 18 49@2x](https://github.com/user-attachments/assets/36e51831-9b3a-42d7-b9d7-24b59b2e1bd9)

Reply:
Actually that should never happen or happen after we move all workflows out of the folder. We should probably add the logic inside the `if (transferToFolderId)` to have a single place where business logic is contain. Else, if we ever expose this functionality via the API or the CLI someone will need to use a mix of this, instead of a calling a single method `delete` which does everything for us.

Reply:
I know this causes the error with the DI, but fixed it [here](https://github.com/n8n-io/n8n/pull/15237/files). That PR contains the suggested changes. Let me know what you think.

Reply:
That PR had some nice suggestions, applied them. Thanks!